hydra-node/test/Hydra/Chain/Direct/Contract/CollectCom.hs

```diff
@@ -131,11 +135,45 @@ genCollectComMutation (tx, utxo) =
         changeRedeemer <- ChangeHeadRedeemer <$> (Head.Close 0 . toBuiltin <$> genHash <*> arbitrary)
         changeDatum <- ChangeHeadDatum <$> (Head.Open <$> arbitrary <*> (toBuiltin <$> genHash))
         pure $ Changes [changeRedeemer, changeDatum]
+    , SomeMutation MutateNumberOfParties <$> do
+        -- NOTE: This also mutates the contestation period becuase we could not
+        -- be bothered to decode/lookup the current one.
+        c <- arbitrary
+        moreParties <- (: healthyOnChainParties) <$> arbitrary
+        pure $
+          Changes
+            [ ChangeHeadDatum $ Head.Initial c moreParties
+            , ChangeOutput 0 $ mutatedPartiesHeadTxOut moreParties
+            ]
     ]
  where
   TxOut collectComOutputAddress collectComOutputValue collectComOutputDatum =
     fromJust $ txOuts' tx !!? 0
 
+  mutatedPartiesHeadTxOut parties =
+    -- NOTE / TODO:
+    --
+    -- This should probably be defined a 'Mutation', but it is different
+    -- from 'ChangeHeadDatum'. The latter modifies the _input_ datum given
+    -- to the script, whereas this one modifies the resulting head datum in
+    -- the output.
+    case collectComOutputDatum of
+      TxOutDatumNone ->
+        error "Unexpected empty head datum"
+      (TxOutDatumHash _ha) ->
+        error "Unexpected hash-only datum"
+      (TxOutDatum sd) ->
+        case fromData $ toPlutusData sd of
+          (Just Head.Open{utxoHash}) ->
+            TxOut
+              collectComOutputAddress
+              collectComOutputValue
+              (mkTxOutDatum Head.Open{parties, utxoHash})
+          (Just st) ->
+            error $ "Unexpected state " <> show st
+          Nothing ->
+            error "Invalid data"
```

Comment:
Hmmm. Yes indeed, I agree with the note.

Reply:
The comment was even copied from below 😅